datasets/dream/dream.py

```diff
@@ -0,0 +1,135 @@
+# coding=utf-8
+# Copyright 2020 HuggingFace Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""DREAM: A Challenge Dataset and Models for Dialogue-Based Reading Comprehension"""
+
+from __future__ import absolute_import, division, print_function
+
+import json
+import logging
+
+import datasets
+
+
+_CITATION = """\
+@article{sundream2018,
+  title={{DREAM}: A Challenge Dataset and Models for Dialogue-Based Reading Comprehension},
+  author={Sun, Kai and Yu, Dian and Chen, Jianshu and Yu, Dong and Choi, Yejin and Cardie, Claire},
+  journal={Transactions of the Association for Computational Linguistics},
+  year={2019},
+  url={https://arxiv.org/abs/1902.00164v1}
+}
+"""
+
+_DESCRIPTION = """\
+DREAM is a multiple-choice Dialogue-based REAding comprehension exaMination dataset. \
+In contrast to existing reading comprehension datasets, DREAM is the first to focus on \
+in-depth multi-turn multi-party dialogue understanding.
+"""
+
+_URL = "https://raw.githubusercontent.com/nlpdata/dream/master/data/"
+_URLS = {
+    "train": _URL + "train.json",
+    "dev": _URL + "dev.json",
+    "test": _URL + "test.json",
+}
+
+
```

Comment:
I don't understand why you have to create a new class rather than using datasets.BuilderConfig directly.